### PR TITLE
Cython based optimization

### PR DESCRIPTION
**Pull request by polm**
This incorporates cythonization of some core modules, along with some other changes, for a significant speedup. In my tests it took [a benchmark](https://github.com/polm/ja-tokenizer-benchmark) from 35s to 10s. 

There's still significant room for improvement, but I wanted to go ahead and get this committed.

### CODE REVIEW

**Comment by polm** [comment]
Not sure why Travis still shows this as queued, but if you click through the tests are passing. 

This is related to #74.

**Comment by polm** [comment]
Just to clarify, I would like to go ahead and get this merged if it looks OK. I can add future improvements in a separate PR.

Let me know if there are any issues with this.

**Comment by sorami** [comment]
Thank you very much for the PR! Let me check, then merge and release a new version.

File: sudachipy/plugin/oov/mecab_oov_plugin.py [hunk 1]
```diff
@@ -32,9 +32,10 @@ def __init__(self):
 
     class OOV:
         def __init__(self):
-            self.left_id = None
-            self.right_id = None
-            self.cost = None
+            #XXX Is this safe?
```
Comment (sorami) [inline sudachipy/plugin/oov/mecab_oov_plugin.py]: This is fine, could you delete this comment line?

**Review by sorami — CHANGES_REQUESTED** [review]

File: sudachipy/plugin/oov/mecab_oov_plugin.py (hunk 1 above)
Comment (polm) [reply sudachipy/plugin/oov/mecab_oov_plugin.py]: Ah, sorry, forgot to remove that after I checked it.

**Comment by sorami** [comment]
Released as v.0.4.6
https://github.com/WorksApplications/SudachiPy/releases/tag/v0.4.6


(The build failed, not on the PyPI yet https://pypi.org/project/SudachiPy/)

**Comment by sorami** [comment]
Not sure why we still get errors with Travis CI.

**Comment by sorami** [comment]
So Travis CI failed to upload to PyPI because now we have Cython build and `bdist_wheel` adds the platform tag `linux_x86_64` which they don't accept.

[Platform compatibility tags — Python Packaging User Guide](https://packaging.python.org/specifications/platform-compatibility-tags/)

I've manually released the source and the wheels for macOS for now; https://pypi.org/project/SudachiPy/#files

**Comment by sorami** [comment]
#125

**Pull request merged**